src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -862,9 +862,9 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 .Select(outputNode => outputNode.OutputData).ToList();
             serializer.Serialize(writer, outputNodeDatas);
 
-            // Nodes
+            // Nodes except for nodes in Transient state
             writer.WritePropertyName("Nodes");
-            serializer.Serialize(writer, ws.Nodes);
+            serializer.Serialize(writer, ws.Nodes.Where(x => x.IsTransient != true);
```

Comment:
This would make sure when node/cluster preview is showing and user press save or shortcut, they dont end up into the dyn